lib/services/lookup.js

```diff
@@ -71,9 +73,76 @@ function lookupServiceFactory(
      * @constructor
      */
     function LookupService () {
+        this.resetNodeIdCache();
     }
 
     /**
+     * Reset all cached Node IDs.
+     */
+    LookupService.prototype.resetNodeIdCache = function () {
+        if (this.nodeIdCache) {
+          this.nodeIdCache.reset();
+          return;
+        }
+
+        this.nodeIdCache = LRU({
+            max: 500
```

Comment:
@benbp should we also be setting a `maxAge` key here in using the cache with the lookup services to limit how stale a cache value can be?